src/functions.jl

```diff
@@ -371,16 +395,20 @@ for T in (:Taylor1, :TaylorN)
             if $T == Taylor1
                 @inbounds c[k] = k * a[k] * c0
             else
-                @inbounds mul!(c[k], k * a[k], c0)
+                @inbounds mul!(c[k], a[k], c0, scalar=k)
             end
             @inbounds for i = 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * a[k-i] * c[i]
                 else
-                    mul!(c[k], (k-i) * a[k-i], c[i])
+                    mul!(c[k], a[k-i], c[i], scalar=k-i)
                 end
             end
-            @inbounds c[k] = c[k] / k
+            if $T == Taylor1
+                @inbounds div!(c, c, k, k)
+            else
+                @inbounds div!(c[k], c[k], k)
+            end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done.